plugins/by-name/ltex-extra/default.nix

```diff
@@ -43,26 +43,60 @@ lib.nixvim.plugins.mkNeovimPlugin {
     ] "Log level.";
   };
 
-  extraConfig = cfg: {
-    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" {
-      when = !config.plugins.lsp.enable;
-      message = ''
-        You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+  extraConfig =
+    cfg:
+    let
+      inherit (config.plugins.lsp.servers) ltex;
+      attachLua = ''
+        require("ltex_extra").setup(${lib.nixvim.toLuaObject cfg.settings})
       '';
-    };
-
-    plugins.lsp = {
-      servers.ltex = {
-        # Enable the ltex language server
-        enable = true;
+    in
+    {
+      warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" [
+        {
+          when = !config.plugins.lsp.enable;
```

Comment:
We're now using `config.plugins.lsp` a couple times. I'd consider having a `lspCfg` binding alongside the `cfg` binding?